src/libopensc/pkcs15-algo.c

```diff
@@ -347,116 +347,120 @@ asn1_free_ec_params(void *params)
 #endif
 #ifdef SC_ALGORITHM_GOSTR3410
-	{ SC_ALGORITHM_GOSTR3410, {{ 1, 2, 643, 2, 2, 19, -1}},
+		{SC_ALGORITHM_GOSTR3410, {{1, 2, 643, 2, 2, 19, -1}},
 			asn1_decode_gostr3410_params,
 			asn1_encode_gostr3410_params,
-			NULL },
+			NULL},
 #endif
 /* We do not support PBES1 because the encryption is weak */
 #ifdef SC_ALGORITHM_PBKDF2
-	{ SC_ALGORITHM_PBKDF2, {{ 1, 2, 840, 113549, 1, 5, 12, -1}},
+		{SC_ALGORITHM_PBKDF2, {{1, 2, 840, 113549, 1, 5, 12, -1}},
 			asn1_decode_pbkdf2_params,
 			asn1_encode_pbkdf2_params,
-			free },
+			free},
 #endif
 #ifdef SC_ALGORITHM_PBES2
-	{ SC_ALGORITHM_PBES2, {{ 1, 2, 840, 113549, 1, 5, 13, -1}},
+		{SC_ALGORITHM_PBES2, {{1, 2, 840, 113549, 1, 5, 13, -1}},
 			asn1_decode_pbes2_params,
 			asn1_encode_pbes2_params,
-			asn1_free_pbes2_params },
+			asn1_free_pbes2_params},
 #endif
 #ifdef SC_ALGORITHM_EC
-	{ SC_ALGORITHM_EC, {{ 1, 2, 840, 10045, 2, 1, -1}},
+		{SC_ALGORITHM_EC, {{1, 2, 840, 10045, 2, 1, -1}},
 			asn1_decode_ec_params,
 			asn1_encode_ec_params,
-			asn1_free_ec_params },
+			asn1_free_ec_params},
 #endif
 /* TODO: -DEE Not clear if we need the next five or not */
 #ifdef SC_ALGORITHM_ECDSA_SHA1
 	/* Note RFC 3279 says no ecParameters */
-	{ SC_ALGORITHM_ECDSA_SHA1, {{ 1, 2, 840, 10045, 4, 1, -1}}, NULL, NULL, NULL},
+		{SC_ALGORITHM_ECDSA_SHA1, {{1, 2, 840, 10045, 4, 1, -1}}, NULL, NULL, NULL},
 #endif
 #ifdef SC_ALGORITHM_ECDSA_SHA224
-/* These next 4 are defined in RFC 5758 */
-	{ SC_ALGORITHM_ECDSA_SHA224, {{ 1, 2, 840, 10045, 4, 3, 1, -1}},
+	/* These next 4 are defined in RFC 5758 */
+		{SC_ALGORITHM_ECDSA_SHA224, {{1, 2, 840, 10045, 4, 3, 1, -1}},
 			asn1_decode_ec_params,
 			asn1_encode_ec_params,
-			asn1_free_ec_params },
+			asn1_free_ec_params},
 #endif
 #ifdef SC_ALGORITHM_ECDSA_SHA256
-	{ SC_ALGORITHM_ECDSA_SHA256, {{ 1, 2, 840, 10045, 4, 3, 2, -1}},
+		{SC_ALGORITHM_ECDSA_SHA256, {{1, 2, 840, 10045, 4, 3, 2, -1}},
 			asn1_decode_ec_params,
 			asn1_encode_ec_params,
-			asn1_free_ec_params },
+			asn1_free_ec_params},
 #endif
 #ifdef SC_ALGORITHM_ECDSA_SHA384
-	{ SC_ALGORITHM_ECDSA_SHA384, {{ 1, 2, 840, 10045, 4, 3, 3, -1}},
+		{SC_ALGORITHM_ECDSA_SHA384, {{1, 2, 840, 10045, 4, 3, 3, -1}},
 			asn1_decode_ec_params,
 			asn1_encode_ec_params,
-			asn1_free_ec_params },
+			asn1_free_ec_params},
 #endif
 #ifdef SC_ALGORITHM_ECDSA_SHA512
-	{ SC_ALGORITHM_ECDSA_SHA512, {{ 1, 2, 840, 10045, 4, 3, 4, -1}},
+		{SC_ALGORITHM_ECDSA_SHA512, {{1, 2, 840, 10045, 4, 3, 4, -1}},
 			asn1_decode_ec_params,
 			asn1_encode_ec_params,
-			asn1_free_ec_params },
+			asn1_free_ec_params},
 #endif
 #ifdef SC_ALGORITHM_EDDSA
 	/* aka Ed25519 */
-	/* RFC 8410, needed to parse/create X509 certs/pubkeys */
-	{ SC_ALGORITHM_EDDSA, {{1, 3, 101, 112, -1}}, NULL, NULL, NULL },
+	/* RFC 8410, needed to parse/create X509 certs/pubkeys  */
+	/* TODO DEE add asn1_decode_ec_params, asn1_encode_ec_params,asn1_free_ec_params */
```

Comment:
Please remove the DEE identifier (git history has information about the author)

Also, please also add this very comment to the SC_ALGORITHM_XEDDSA block below

Reply:
Removed comment and added the asn1_decode_ec_params, asn1_encode_ec_params,asn1_free_ec_params as needed.

Reply:
And the asn1_decode_ec_params, asn1_encode_ec_params,asn1_free_ec_params  where added for SC_ALGORITHM_XEDDSA.